test/with_dummyserver/test_socketlevel.py

```diff
@@ -618,6 +624,57 @@ def test_httplib_headers_case_insensitive(self):
         self.assertEqual(HEADERS, dict(r.headers.items())) # to preserve case sensitivity
 
 
+class TestBrokenHeaders(SocketDummyServerTestCase):
+    def setUp(self):
+        if issubclass(httplib.HTTPMessage, MimeToolMessage):
+            raise SkipTest('Header parsing errors not available')
+
+        super(TestBrokenHeaders, self).setUp()
+
+    def _test_broken_header_parsing(self, headers):
+        handler = create_response_handler((
+           b'HTTP/1.1 200 OK\r\n'
+           b'Content-Length: 0\r\n'
+           b'Content-type: text/plain\r\n'
+           ) + b'\r\n'.join(headers) + b'\r\n'
+        )
+
+        self._start_server(handler)
+        pool = HTTPConnectionPool(self.host, self.port, retries=False)
+
+        with LogRecorder() as logs:
```

Comment:
Can you remind me why this is better than `with warnings.catch_warnings(record=True) as w:`?

I recall somebody mentioned some shortcomings but I can't find what they were.